src/utils/template_handler.rs

```diff
@@ -101,7 +117,12 @@ pub(crate) fn reload_template(name: String, strict: bool, reset: bool) -> Status
 }
 
 /// Load a collection of templates from a remote repository
-pub(crate) fn load_remote_template_collection(path: &str, url: &str, force: bool) -> Status {
+pub(crate) fn load_remote_template_collection(
+    path: &str,
+    url: &str,
+    force: bool,
+    url_type: &URLType,
```

Comment:
Each time this function is called the URL has to be checked to determine the right type. This adds code repetition. Therefore I would suggest to handle the URL type within this function and also check for if the URL is not supported. In this case return a status with an error and the caller then has to check that status as well.

Reply:
Thats an amazing point @cophilot , added the url check inside this function, so that this function can become more resuable with lesser code repition. Also, added the check inside for an invalid url and added the test for the same.